asf_search/WKT/validate_wkt.py

```diff
@@ -0,0 +1,330 @@
+                    raise ASFWKTError(f'WKT string: \"{aoi_shape.wkt}\" is a self intersecting polygon')
+   
+            raise ASFWKTError(f'WKT string: \"{aoi_shape.wkt}\" is not a valid WKT string')
+    
+    simplified = _simplify_geometry(aoi_shape)
+    
+    return simplified
+
+
+def _search_wkt_prep(shape: BaseGeometry):
+
+    if isinstance(shape, (Point, LineString)):
+        return shape
+
+    if isinstance(shape, BaseMultipartGeometry) :
+        output = []
+        for geom in shape.geoms:
+            if isinstance(geom, Polygon):
+                output.append(orient(geom))
+            else:
+                output.append(geom)
+
+        if isinstance(shape, MultiPolygon):
+            return MultiPolygon(output)
+        if isinstance(shape, MultiLineString):
+            return MultiLineString(output)
+        if isinstance(shape, MultiPoint):
+            return MultiPoint(output)
+        if isinstance(shape, GeometryCollection):
+            return GeometryCollection(output)
+                         
+    
+    if isinstance(shape, Polygon):
+        return orient(shape, sign=1.0)
+    
+    raise ASFWKTError(f'The provided WKT is not a valid type. Valid WKT types include \"(Multi-)Point\", \"(Multi-)LineString\", \"(Multi-)Polygon\", and \"GeometricCollections\"')
+
+
+def _simplify_geometry(geometry: BaseGeometry) -> BaseGeometry:
+    """
+    param geometry: AOI Shapely Geometry to be prepped for CMR 
+    prepares geometry for CMR by:
+        1. Flattening any nested multi-part geometry into single collection
+        2. clamping latitude +/-90, unwrapping longitude +/-180, removing coordinate dimensions higher than 2 (lon,lat)
+        3. Merging any overlapping shapes
+        4. convex-hulling the remainder into a single shape
+        4. simplifing until the shape has <= 300 points, with no point closer than 0.00001
+        5. Orienting vertices in counter-clockwise winding order
+    returns: geometry prepped for CMR
+    """
+    flattened = _flatten_multipart_geometry(geometry)
+    clamped, clamp_report = _get_clamped_geometry(flattened)
+    merged, merge_report = _merge_overlapping_geometry(clamped)
+    convex, convex_report = _get_convex_hull(merged)
+    simplified, simplified_report = _simplify_aoi(convex)
+    reoriented, reorientation_report = _counter_clockwise_reorientation(simplified)
+
+    dimension_report = RepairEntry(
+        report_type="'type': 'EXTRA_DIMENSION'", 
+        report="'report': Only 2-Dimensional area of interests are supported (lon/lat), higher dimension coordinates will be ignored"
+        ) if geometry.has_z else None
+
+    if convex_report != None:
+        merge_report = None
+
+    repair_reports = [dimension_report, merge_report, convex_report, *clamp_report, *simplified_report, reorientation_report]   
+    for report in repair_reports:
+        if report is not None:
+            logging.info(f"{report.report_type}\n\t{report.report}")
```

Comment:
Looks like by default, the info level isn't shown to the console, only warning+. This doesn't feel like a warning though. Can/should we change the default print down to info? The logging config would go in the first `__init__.py` I guess?

I can't figure out how to see output, so I can't check any repair messages.

Reply:
Ideally we'll want users to set their own logging handlers